### ISSUE TITLE
Refactor MesosClusterConfig API to be more fluent

### ISSUE THREAD

**Issue by philwinder** [ContainerSolutions/minimesos#144]
Over time, we've had to had lots of nasty helper methods to the MesosCluster API. Like `slaveResources` and `mesosMasterImage`. This is only going to continue to increase.

A better method would be to allow the user to specify the architecture of the cluster, and provide an optional definition for that container. For example:
```
MesosClusterConfig.withMaster().withMaster(AbstractContainer myMaster).withSlave().withSlave(AbstractContainer mySlave).withZookeeper().withContainer(AbstractContainer newContainer);
```
This way, the user can override the MesosMaster/Slave to do whatever they want, without having to alter the minimesos API.

**Comment by sadovnikov** [comment]
I wonder how the framework is going to work on a real Mesos cluster. There won't be much control over slaves on real clusters. And there will be challenges of discovering the Scheduler and Executors.
Fixing ``mesos-hello-world`` today, I ended up doing inspection of containers of executors - this was the only feasible way to find out IP addresses of deployed web servers. This seems to be very wrong, but I did not have a better idea. 

Should jars be pulled from external repository, something like what happens in install script of ``minimesos``?
